src/Extensions.cs

```diff
@@ -370,9 +380,19 @@ public static bool HasInterface(this IModelType type)
         /// <returns></returns>
         public static string GetInterfaceName(this IModelType type, bool includePkgName = false)
         {
+            // this function is called in a *LOT* of places so is perf sensitive
+            if (!s_interfaceNames.ContainsKey(type))
+            {
+                var baseName = $"Basic{type.Name}";

```

Comment:
nit: interfaceName instead of baseName ?

Reply:
Will fix.  I also have another pending fix for perf, will include in that.